src/step.js

```diff
@@ -53,10 +57,12 @@ class Step {
                 step.context = this.context
 
                 let ret = await step.run()
-
+  
                 this._auditTrail.steps.push(step.auditTrail)
 
                 if (ret.isErr) return ret
+
+                if(step.context._stopExecution ) break
```

Comment:
Would be nice to have a extra audit info making it explicit that the UC as stopped by calling `.stop()`

Reply:
More than one people judge that is a good idea and I dont see why not. Already send a new commit with wasStopped flag.

Reply:
``` javascript
it('should audit with Ok result', async () => {
                //given
                const uc = givenAMultiStepUseCase()
                //when
                const ret = await uc.run()
                //then
                assert.deepStrictEqual(uc.auditTrail, {
                    type: 'use case',
                    description: 'A use case',
                    transactionId: uc._mainStep._auditTrail.transactionId,
                    elapsedTime: uc._mainStep._auditTrail.elapsedTime,
                    return: { Ok: { sucess: true }},
                    steps: [
                        { type: 'step', description: 'step 1', return: { Ok: '' }, elapsedTime: uc._auditTrail.steps[0].elapsedTime },
                        { type: 'step', description: 'step 2', wasStopped: true, return: { Ok: '' }, elapsedTime: uc._auditTrail.steps[1].elapsedTime },
                    ]
                })
```

I saw you put the stop flag inside the step. It would not be better to do something like this:

``` javascript

{
                    type: 'use case',
                    description: 'A use case',
                    transactionId: uc._mainStep._auditTrail.transactionId,
                    elapsedTime: uc._mainStep._auditTrail.elapsedTime,
                    stopped: true,
                    return: { Ok: { sucess: true }},
                    steps: [
                        { type: 'step', description: 'step 1', stopped: false, return: { Ok: '' }, elapsedTime: uc._auditTrail.steps[0].elapsedTime },
                        { type: 'step', description: 'step 2', stopped: true, return: { Ok: '' }, elapsedTime: uc._auditTrail.steps[1].elapsedTime },
                    ]
}

```

Put the variable stopped explicity in the usecase, and also indicate wich step was stopped.

Reply:
> Put the variable stopped explicity in the usecase

I will accept the PR as it is and leave this as a improvement.

Reply:
I dont see that like a improvement, I believe I stopped is a attribute from the step, not the UC.